src/nvim/api/buffer.c

```diff
@@ -125,7 +125,7 @@ Integer nvim_buf_line_count(Buffer buffer, Error *err)
 ///             - on_bytes: Lua callback invoked on change.
 ///               This callback receives more granular information about the
 ///               change compared to on_lines.
-///               Return `true` to detach.
+///               Return a truthy value (not `false` or `nil`) to detach.
```

Comment:
There is another mention at `on_lines`

Reply:
73c87c5aead2c21eec0889f9d86952a0bee4f8aa